[Fleet] Document agent endpoints (#47146)
elastic/kibana: docs/api/fleet/agent_api_enroll.asciidoc

[[fleet-agent-enroll]]

> === Fleet agent enroll API

Reviewer: ```suggestion
=== Enroll Fleet agent API
```

++++

> <titleabbrev>Enroll a new agent</titleabbrev>

Reviewer: ```suggestion
<titleabbrev>Enroll agent</titleabbrev>
```

++++

[[fleet-agent-enroll-request]]
==== Request

`POST /api/fleet/agents/enroll` 

==== Headers

`kbn-fleet-enrollment-token`::
  (Required, string) A fleet enrollment token.

[[fleet-agent-enroll-request-body]]
==== Request body

`sharedId`::
  (Optional, string) An ID for the agent.

`metadata`::

>   (Optional, object) Objects with `local`, and `userProvided` properties that contains the metadata for an agent. Metadata are a dictonary of strings (example: `"local": { "os": "macos" }`).

Reviewer: ```suggestion
  (Optional, object) Objects with `local` and `userProvided` properties that contain the metadata for an agent. The metadata is a dictionary of strings (example: `"local": { "os": "macos" }`).
```

[[fleet-agent-enroll-request-codes]]
==== Response code

`200`:: 
    Indicates a successful call.
    
[[leet-agent-enroll-example]]
==== Example

[source,js]
--------------------------------------------------

> POST /api/fleet/agents/enroll

Reviewer: Same comment as before re: formatting (possibly caused by how aciidoctor preserves tabs):

![image](https://user-images.githubusercontent.com/14206422/66243820-e615ba80-e6ba-11e9-8ede-77162ff5a6bd.png)
